numpy/core/src/multiarray/flagsobject.c

```diff
@@ -755,8 +740,8 @@ arrayflags_richcompare(PyObject *self, PyObject *other, int cmp_op)
     }
 
     if (PyObject_TypeCheck(other, &PyArrayFlags_Type)) {
-        cmp = arrayflags_compare((PyArrayFlagsObject *)self,
-                                 (PyArrayFlagsObject *)other);
+        cmp = ((PyArrayFlagsObject*) self)->flags !=
```

Comment:
Now that cmp is a boolean, perhaps:
```suggestion
        is_equal = ((PyArrayFlagsObject*) self)->flags ==
```
And replace `cmp == 0` with `is_equal`, `cmp != 0` with `!is_equal`

Reply:
Or a full rewrite, also taking advantage of the fact there is no longer a need to raise `TypeError`:

```C
static PyObject*
arrayflags_richcompare(PyObject *self, PyObject *other, int cmp_op)
{
    if (!PyObject_TypeCheck(other, &PyArrayFlags_Type)) {
        return Py_RETURN_NOTIMPLEMENTED;
    }

    npy_bool eq = ((PyArrayFlagsObject*) self)->flags ==
                   ((PyArrayFlagsObject*) other)->flags;

    if (cmp_op == Py_EQ) {
        return PyBool_FromLong(eq)
    }
    else if (cmp_op == Py_NE) {
        return PyBool_FromLong(neq)
    }
    else {
        Py_RETURN_NOTIMPLEMENTED;
    }
}
```

Reply:
Newbie question, why doesn't this require a Py_INCREF on the returned boolean?
Is it because [Cpython calls INCREF on the reference for you](https://github.com/python/cpython/blob/master/Objects/boolobject.c#L36)? If so am I just supposed to know that happens? the [api doc](https://docs.python.org/3.7/c-api/bool.html) doesn't mention anything.

Reply:
Most python functions return a new reference. These type of functions have to. The only functions that do not sometimes are things that give you back something already inside a dictionary or list or so that you already own.

Reply:
Thanks for the explanation, I'll look around the source some more and see if I can internalize this.